builtin/providers/google/resource_sql_user.go

```diff
@@ -95,32 +103,44 @@ func resourceSqlUserRead(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	name := d.Get("name").(string)
-	instance := d.Get("instance").(string)
+	instanceAndName := strings.SplitN(d.Id(), ".", 2)
+	if len(instanceAndName) != 2 {
+		return errors.New(
+			fmt.Sprintf(
+				"Wrong number of arguments when specifying imported id. Expected: 2.  Saw: %d. Expected Input: $INSTANCENAME.$SQLUSERNAME Input: %s",
+				len(instanceAndName),
+				d.Id()))
+	}
+
+	instance := instanceAndName[0]
+	name := instanceAndName[1]
 
 	users, err := config.clientSqlAdmin.Users.List(project, instance).Do()
 
 	if err != nil {
 		return handleNotFoundError(err, d, fmt.Sprintf("SQL User %q in instance %q", name, instance))
 	}
 
-	found := false
-	for _, user := range users.Items {
-		if user.Name == name {
-			found = true
+	var user *sqladmin.User
+	for _, currentUser := range users.Items {
+		if currentUser.Name == name {
+			user = currentUser
 			break
 		}
 	}
 
-	if !found {
+	if user == nil {
 		log.Printf("[WARN] Removing SQL User %q because it's gone", d.Get("name").(string))
 		d.SetId("")
 
 		return nil
 	}
 
-	d.SetId(name)
+	d.SetId(fmt.Sprintf("%s.%s", instance, name))
```

Comment:
It's generally not a good idea to call `SetId` outside of a `CREATE` method. I realize this isn't your addition, but do you know off hand why this is here? Can it be removed and have the resource work as expected?